packages/core/onClickOutside/directive.ts

```diff
@@ -1,19 +1,26 @@
-import type { FunctionDirective } from 'vue-demi'
+import { directiveHooks } from '@vueuse/shared'
+import type { ObjectDirective } from 'vue-demi'
 import { onClickOutside } from '.'
+import type { OnClickOutsideHandler, OnClickOutsideOptions } from '.'
 
-const handler = (): FunctionDirective<any, <E = PointerEvent>(evt: E) => void> => {
-  let stop = null as unknown as ReturnType<typeof onClickOutside>
-  return (el, binding) => {
-    if (stop) {
-      stop()
-      stop = onClickOutside(el, binding.value)
-      return
+export const vOnClickOutside: ObjectDirective<
+HTMLElement,
+OnClickOutsideHandler | [(evt: any) => void, OnClickOutsideOptions]
```

Comment:
Unfortunately, I haven't been able to conditionally switch between `PointerEvent | FocusEvent` and `PointerEvent` based on `detectIframe` value like I did in `index.ts:38`.